librepo/metadata_downloader.c

```diff
@@ -69,12 +70,14 @@ lr_metadatatarget_new2(LrHandle *handle,
                        LrProgressCb progresscb,
                        LrMirrorFailureCb mirrorfailure_cb,
                        LrEndCb endcb,
+                       char *gnupghomedir,
                        GError **err)
 {
     LrMetadataTarget *target = lr_metadatatarget_new(handle, NULL, NULL, cbdata, err);
     target->progresscb = progresscb;
     target->mirrorfailurecb = mirrorfailure_cb;
     target->endcb = endcb;
+    target->gnupghomedir = g_strdup(gnupghomedir);
```

Comment:
Structure target uses target->chunk to store strings. 
So be consistent, use "target->gnupghomedir = g_string_chunk_insert(target->chunk, gnupghomedir)"
if you want use g_strdup() then you have to add "g_free(target->gnupghomedir);" to lr_metadatatarget_free()